fido/src/main/java/com/yubico/yubikit/fido/client/extensions/HmacSecretExtension.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (C) 2024 Yubico.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yubico.yubikit.fido.client.extensions;
+
+import static com.yubico.yubikit.core.internal.codec.Base64.fromUrlSafeString;
+import static com.yubico.yubikit.core.internal.codec.Base64.toUrlSafeString;
+
+import com.yubico.yubikit.core.application.CommandException;
+import com.yubico.yubikit.core.internal.Logger;
+import com.yubico.yubikit.core.util.Pair;
+import com.yubico.yubikit.core.util.StringUtils;
+import com.yubico.yubikit.fido.ctap.ClientPin;
+import com.yubico.yubikit.fido.ctap.Ctap2Session;
+import com.yubico.yubikit.fido.ctap.PinUvAuthProtocol;
+import com.yubico.yubikit.fido.webauthn.AttestationObject;
+import com.yubico.yubikit.fido.webauthn.AuthenticatorData;
+import com.yubico.yubikit.fido.webauthn.Extensions;
+import com.yubico.yubikit.fido.webauthn.PublicKeyCredentialCreationOptions;
+import com.yubico.yubikit.fido.webauthn.PublicKeyCredentialDescriptor;
+import com.yubico.yubikit.fido.webauthn.PublicKeyCredentialRequestOptions;
+
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+public class HmacSecretExtension extends Extension {
+
+    private static final org.slf4j.Logger logger = LoggerFactory.getLogger(HmacSecretExtension.class);
+    private static final int SALT_LEN = 32;
+
+    public HmacSecretExtension() {
+        super("hmac-secret");
+    }
```

Comment:
I think it might make sense to do as in python-fido2, where we have a boolean that determines if hmac-secret should be processed or not (if not, PRF still works) and default to disabling it. The rationale being that the "default" set of extensions should mainly follow what most browsers do, and hmac-secret is not exposed to RPs via the browser.

Reply:
Updated in [70ea7d0](https://github.com/Yubico/yubikit-android/pull/161/commits/70ea7d07e9757cbc07626643bd800574826b31b0).

hmac-secret is only available if allowed through the `HmacSecretExtension(boolean allowHmacSecret)` constructor. There is also a parameter less constructor which creates instance with not-allowed hmac-secret.

Updated device tests accordingly.